core/src/main/java/com/amplitude/core/utilities/EventsFileManager.kt

```diff
@@ -122,50 +146,90 @@ class EventsFileManager(
         val firstHalfFile = File(directory, "$fileName-1.tmp")
         val secondHalfFile = File(directory, "$fileName-2.tmp")
         val splitStrings = events.split()
-        writeToFile(splitStrings.first, firstHalfFile)
-        writeToFile(splitStrings.second, secondHalfFile)
+        writeEventsToSplitFile(splitStrings.first, firstHalfFile)
+        writeEventsToSplitFile(splitStrings.second, secondHalfFile)
         this.remove(filePath)
     }
 
-    suspend fun getEventString(filePath: String): String = readMutex.withLock {
-        // Block one time of file reads if another task has read the content of this file
-        if (filePathSet.contains(filePath)) {
-            filePathSet.remove(filePath)
-            return ""
-        }
-        filePathSet.add(filePath)
-        File(filePath).bufferedReader().use<BufferedReader, String> {
-            return it.readText()
+    suspend fun getEventString(filePath: String): String =
+        readMutex.withLock {
+            // Block one time of file reads if another task has read the content of this file
+            if (filePathSet.contains(filePath)) {
+                filePathSet.remove(filePath)
+                return@withLock ""
+            }
+            filePathSet.add(filePath)
+            File(filePath).bufferedReader().use<BufferedReader, String> {
+                val content = it.readText()
+                val isCurrentVersion = content.endsWith("\n")
+                if (isCurrentVersion) {
+                    // handle current version
+                    val events = JSONArray()
+                    content.split("\n").forEach {
+                        if (it.isNotEmpty()) {
+                            try {
+                                events.put(JSONObject(it))
+                            } catch (e: JSONException) {
+                                logger.error("Failed to parse event: $it")
```

Comment:
It feels unsafe to drop events in the SDK without us being aware of it. Can we still send these events to the server on a different field and have an alert on the server side to alert us when we run into malformed data on the SDK?

Reply:
Let's discuss it later. Is there a field now could be a good fit?

Reply:
Added the basic diagonostircs, feel free to let me know if we would like to put the interface in a different way. Thanks. cc @justin-fiedler